src/charm.py

```diff
@@ -867,6 +874,29 @@ def _install_snap_packages(self, packages: List[str]) -> None:
                 )
                 raise
 
+    def _patch_snap_seccomp_profile(self) -> None:
+        """Patch snap seccomp profile to allow chmod on pgBackRest restore code.
+
+        This is needed due to https://github.com/pgbackrest/pgbackrest/issues/2036.
+        """
+        subprocess.check_output(
+            [
+                "sed",
+                "-i",
+                "-e",
+                "$achown",
+                "/var/lib/snapd/seccomp/bpf/snap.charmed-postgresql.patroni.src",
+            ]
+        )
+        subprocess.check_output(
+            [
+                "/usr/lib/snapd/snap-seccomp",
+                "compile",
+                "/var/lib/snapd/seccomp/bpf/snap.charmed-postgresql.patroni.src",
+                "/var/lib/snapd/seccomp/bpf/snap.charmed-postgresql.patroni.bin",
+            ]
+        )
```

Comment:
Workaround to enable the restore of a backup work (more details in the following issue: https://github.com/pgbackrest/pgbackrest/issues/2036).